### PR TITLE
system variable: add the switch control for outer join reorder

### PR DESCRIPTION
**Pull request by winoros**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the description for new system variable to control the outer join reorder.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Reminiscent — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -785,6 +785,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 >
 > 该变量只有在默认值 `OFF` 时，才算是安全的。因为设置 `tidb_enable_noop_functions=1` 后，TiDB 会自动忽略某些语法而不报错，这可能会导致应用程序出现异常行为。例如，允许使用语法 `START TRANSACTION READ ONLY` 时，事务仍会处于读写模式。
 
+### `tidb_enable_outer_join_reorder` <span class="version-mark">从 v6.1 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:788]:
```suggestion
### `tidb_enable_outer_join_reorder` <span class="version-mark">从 v6.1.0 版本开始引入</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -785,6 +785,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- TiDB 在 v6.1 中对 Join Reorder 算法进行了改进，使得算法可以将 Outer Join 也考虑在内。该变量用于控制这个行为，默认开启。
```
Comment (TomShawn) [inline system-variables.md:793]: 如果从旧版本升上来也是默认开启吗？
```suggestion
- 自 v6.1.0 起，TiDB 的 [Join Reorder 算法](/join-reorder.md) 开始支持 Outer Join。该变量用于控制这个支持行为，默认开启。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**